patchify

```diff
@@ -2025,18 +2036,23 @@ def remove_patch(target_patch_id, skip_services, ssh_details, action):
     )
 
 
-def generate_patch_id(patch_name, definition_md5sum):
+def generate_patch_id(patch_name, definition_hash):
     """Generate the ID of a patch. This method should match that used in the
     central registry update tool."""
-    return patch_name + '-' + definition_md5sum
+    return patch_name + '-' + definition_hash
 
 
 def add_patch_to_registry(registry, patch_id, timestamp,
                           patch_name, patch_path, patch_definition,
                           manager_connection_string, ssh_port, keyfile):
     current_patches = registry.get('current_patches', [])
 
-    affected_files = patch_definition['md5sums_after'].keys()
+    if 'sha256sums_after' in patch_definition:
+        affected_files = patch_definition['sha256sums_after'].keys()
+        modified_files = patch_definition['sha256sums_after']
+    else:
+        affected_files = patch_definition['md5sums_after'].keys()
```

Comment:
might as well drop the `.keys()` for symmetry here

Reply:
They're deliberately with just .keys, because we check against them a couple of lines further on. modified_files needs the full details.